src/sentry/sentry_metrics/indexer/ratelimiters.py

```diff
@@ -180,3 +189,65 @@ def check_write_limits(self, use_case_id: UseCaseKey, keys: KeyCollection) -> Ra
 
 
 writes_limiter = WritesLimiter()
+
+
+class WritesLimitingStringIndexerDecorator(StringIndexer):
```

Comment:
nit: About the naming, is this actually a decorator? Seems like it is not. How about `WritesLimitingStringIndexerWrapper` or something?